src/services/refactors/inferFunctionReturnType.ts

```diff
@@ -59,7 +59,9 @@ namespace ts.refactor.inferFunctionReturnType {
         if (isInJSFile(context.file) || !refactorKindBeginsWith(inferReturnTypeAction.kind, context.kind)) return;
 
         const token = getTokenAtPosition(context.file, context.startPosition);
-        const declaration = findAncestor(token, isConvertibleDeclaration);
+        const declaration = findAncestor(token, n =>
+            isBlock(n) || isParameter(n) || n.parent && isArrowFunction(n.parent) && (n.kind === SyntaxKind.EqualsGreaterThanToken || n.parent.body === n) ? "quit" :

```

Comment:
I would want the refactor to show up if the cursor were anywhere inside the parameters (I think just removing the quit `isParameter` should do that). It would be pretty awkward to have to select the whole list to see it.